### DISCUSSION TITLE
HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL not working?

### DISCUSSION THREAD

**Discussion by Zerebokep** [hasura/graphql-engine#8756] [Question] [answered]
The documentation says HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL defines a base url for actions, but I still have to set a base url, even when that env var is defined.

How should we use that env variable?

**Comment by Zerebokep** [comment]
Anyone?

**Comment by meetzaveri** [comment, marked as answer]
Hello,

Thanks for asking. 

`HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL` is used to set base webhook url for action handler. This means that if you don't want to pass URL explicitly in action handler, then you can just set it as value of that env var and pass that env var in action webhook handler

### Setting up env var:
For Hasura CE (open source version):
You can set them in `docker-compose.yaml` file under your graphql-engine image environment. Just add key as `HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL` and pass URL as your value to it.

For Hasura Cloud:
You can directly set them under Env Vars section under your project settings. Cloud dashboard -> Your project Settings -> Env Vars -> New Env Var.

### Implementing in action:
You can use it in your webhook handler input while creating action by passing this string value as `{{HASURA_GRAPHQL_ACTIONS_HANDLER_WEBHOOK_BASEURL}}`. This will use the URL that you had defined in previously in `docker-compose.yaml` file. Note that, you'll need to put `{{` opening and `}}` closing curly braces as described in that string
